jdaviz/core/template_mixin.py

```diff
@@ -1314,13 +1318,80 @@ def add_results_from_plugin(self, data_item):
             self.app.data_collection.remove(self.app.data_collection[self.label])
 
         self.app.add_data(data_item, self.label)
+        if ref_data is not None:
+            self.link_results_from_plugin(ref_data)
 
         if self.add_to_viewer_selected != 'None':
             # replace the contents in the selected viewer with the results from this plugin
             # TODO: switch to an instance/classname check?
             self.app.add_data_to_viewer(self.viewer.selected_id,
                                         self.label, clear_other_data=replace)
 
+    def link_results_from_plugin(self, ref_data):
```

Comment:
I do not see any logic to remove outdated links. So let's say someone calculates Moment 0, and then someone press the button again (which I think overwrites the result), then new links are added but old links remain as well? How would Glue consolidate duplicate links, or will Glue be smart enough?

Reply:
There is no logic for removing duplicate links, I don't know if that functionality already exists in glue. I'll bring it up at tomorrow's SME tag up.